lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -412,14 +412,18 @@ public static function get_merged_data( $settings = array(), $origin = 'user' )
 
 		if (
 			! get_theme_support( 'experimental-link-color' ) && // link color support needs the presets CSS variables regardless of the presence of theme.json file.
-			! WP_Theme_JSON_Resolver_Gutenberg::theme_has_support()
+			! self::theme_has_support()
 		) {
 			return $result;
 		}
 
 		$theme_support_data = WP_Theme_JSON_Gutenberg::get_from_editor_settings( $settings );
 		$result->merge( self::get_theme_data( $theme_support_data ) );
 
+		$filter_data = new WP_Theme_JSON_Gutenberg( apply_filters( 'theme_json_resolver_merged_data', $result->get_raw_data() ) );
```

Comment:
I'd avoid adding filters as of now, as per https://github.com/WordPress/gutenberg/pull/34784/files#r709896696 but also [this](https://github.com/WordPress/gutenberg/issues/27504#issuecomment-858761483) and [this](https://github.com/WordPress/gutenberg/pull/28687#issuecomment-920241825) may be good context.

https://github.com/WordPress/gutenberg/pull/34843 is promising. If we decide to go for something like that, I see the filters belonging to those functions instead of this inner classes.